### PR TITLE
L1 fits write

### PR DESCRIPTION
**Pull request by nicHoch**

### CODE REVIEW

File: stixcore/products/level0/quicklook.py [hunk 1]
```diff
@@ -49,11 +49,14 @@ def __init__(self, *, service_type, service_subtype, ssid, control, data,
...
-                                          + self.control['integration_time'][-1] / 2)
-        self.obs_avg = self.obs_beg + (self.obs_end - self.obs_beg) / 2
+        try:
+            self.obs_beg = SCETime.from_float(self.data['time'][0]
+                                              - self.control['integration_time'][0] / 2)
+            self.obs_end = SCETime.from_float(self.data['time'][-1]
+                                              + self.control['integration_time'][-1] / 2)
+            self.obs_avg = self.obs_beg + (self.obs_end - self.obs_beg) / 2
+        except ValueError:
+            pass
```
Comment (samaloney) [inline stixcore/products/level0/quicklook.py:59]: Hum if we get a value error here I think we are in bad shape do you know what was causing it?

File: stixcore/products/level1/quicklook.py [hunk 2]
```diff
@@ -1,142 +1,37 @@
...
-        self.service_subtype = service_subtype
-        self.ssid = ssid
-        self.control = control
-        self.data = data
-
-        energies = kwargs.get('energies', False)
-        if not energies:
-            energies = self.get_energies()
-        self.energies = energies
+class QLProduct(QLProductL0):
```
Comment (samaloney) [inline stixcore/products/level1/quicklook.py:18]: it's a bit odd to have the baseclass like this maybe we should rename, but not entirely sure why this was done so maybe it's fine?

**Review by samaloney — APPROVED** [review]
Looks good to me. We now have a lot of references to idbs being passed around are they actual idb objects or the version labels?

File: stixcore/products/level0/quicklook.py (hunk 1 above)
Comment (nicHoch) [reply stixcore/products/level0/quicklook.py:59]: yes this was about my question about the timing conversions
some old test fits file had SCETime other "real" time within

File: stixcore/products/level1/quicklook.py (hunk 2 above)
Comment (nicHoch) [reply stixcore/products/level1/quicklook.py:18]: the QLProductL0 is just an import rename. Basically it means that this Level1QLProduct inherits from L0QLProduct

File: stixcore/processing/engineering.py [hunk 3]
```diff
@@ -168,8 +168,12 @@ def raw_to_engineering_product(product, idbm):
 
         # replace the old column with the converted
         product.data[col] = product.data[CCN]
+        product.data[col].meta = product.data[CCN].meta
         # delete the generic column for conversion
         del product.data[CCN]
+        # delete the calibration key from meta as it is now processed
+        del product.data[col].meta["PCF_CURTX"]
+
         assert c == len(product.data)
```
Comment (samaloney) [inline stixcore/processing/engineering.py:177]: asserts are not guaranteed to be evaluated in need and `if else` or `try except`

**Pull request merged**